bgpd/bgpd.c

```diff
@@ -2431,8 +2431,13 @@ static void peer_group2peer_config_copy(struct peer_group *group,
 	if (!CHECK_FLAG(peer->flags_override, PEER_FLAG_LOCAL_AS))
 		peer->change_local_as = conf->change_local_as;
 
-	/* TTL */
-	peer->ttl = conf->ttl;
+	/* TTL ... Retain the Peer's TTL if it is not configured in the peer group.  It'll be over-ridden later. */
+	if ((conf->ttl == BGP_DEFAULT_TTL) && (peer->ttl != BGP_DEFAULT_TTL)) {
```

Comment:
what happens if peer group has ebgp multihop configured with the value of 2 for instance?

Reply:
Then the TTL 2 gets applied. See sample test output below:

sonic(config-router)# do sho ru
Building configuration...

Current configuration:
!
frr version 7.0.1-sonic
frr defaults traditional
hostname sonic
!
router bgp 10
bgp router-id 50.1.1.10
neighbor pg peer-group
neighbor 50.1.1.11 remote-as 11
neighbor 50.1.1.11 peer-group pg
neighbor 50.1.1.11 ebgp-multihop 255
neighbor 50.1.1.11 update-source 2.2.2.2
!
address-family ipv4 unicast
network 8.0.10.0/24
redistribute connected
neighbor 50.1.1.11 prefix-list nag in
exit-address-family
!
ip prefix-list nag seq 5 deny 1.0.1.0/24
ip prefix-list nag seq 10 permit any
!
line vty
!
end
sonic(config-router)# neighbor pg ebgp-multihop 222
sonic(config-router)# do sho ru
Building configuration...

Current configuration:
!
frr version 7.0.1-sonic
frr defaults traditional
hostname sonic
!
router bgp 10
bgp router-id 50.1.1.10
neighbor pg peer-group
neighbor pg ebgp-multihop 222
neighbor 50.1.1.11 remote-as 11
neighbor 50.1.1.11 peer-group pg

Reply:
If you have a configured peer without PG with TTL of 255, and PG has TTL of 2. Now you add a peer to PG which has TTL 2. What should happen?

Reply:
TTL 2 should get configured.

Reply:
isn't your change just a simplified version of:
```
if (conf->ttl > 1) {
    peer->ttl = conf->ttl;
```
?